data_science_pipeline/utils/bq.py

```diff
@@ -248,3 +248,66 @@ def run_query_and_save_to_table(  # pylint: disable=too-many-arguments
     if LOGGER.isEnabledFor(logging.DEBUG):
         sample_result = list(islice(result, 3))
         LOGGER.debug("sample_result: %s", sample_result)
+
+
+def does_bq_table_exist(
+        client: Client,
+        project: str,
+        dataset: str,
+        table: str
+):
+    table_id = ".".join([project, dataset, table])
+    try:
+        client.get_table(table_id)
+        LOGGER.info("Table %s already exists.", table_id)
+        return True
+    # pylint disable-broad-except
+    except Exception as e:
+        LOGGER.info("Table not exist. %s", e)
+        return False
+
+
+def create_or_replace_bq_view(
+        query: str,
+        project: str,
+        dataset: str,
+        view_name: str
+):
+    client = get_client(project_id=project)
+    view_id = ".".join([project, dataset, view_name])
+    view = bigquery.Table(view_id)
+    view.view_query = query
+    if not does_bq_table_exist(client, project, dataset, view_name):
+        view = client.create_table(view)
+        LOGGER.info("Created %s: %s", view.table_type, str(view.reference))
+    else:
+        client.delete_table(view_id, not_found_ok=True)
```

Comment:
I would generally try to avoid deleting a table as part of the general flow (unless it is a cleanup task of course). That is because it is not atomic.

You could look at the
[bigquery-views-manager/](https://github.com/elifesciences/bigquery-views-manager/blob/develop/bigquery_views_manager/update_views.py) that we use for the BigQuery views project. It does indeed use the `CREATE OR REPLACE TABLE` you opted for below.

Reply:
Maybe we could even use the `bigquery-views-manager` project to re-use some code?

Reply:
It is actually just deleting the view which is not in use. It is just for seeing the code via bigquery interface without looking at github. Table and materialised view is exist all the time.

Reply:
Personally I would always prefer an atomic operation where possible (and re-use code where we can). In this case we already have used a method that is atomic for the BigQuery Views. I don't quite see the advantage of not using that.

But if you really want to stick to this, then I would indicating in the name that it is not atomic in the function name to avoid using for other purposes.